addon/bootstrap.js

```diff
@@ -132,29 +141,24 @@ function createLog(name, levelWord) {
   return L;
 }
 
-async function chooseVariation() {
-  let toSet, source;
-  const sample = studyUtils.sample;
-
-  if (studyConfig.variation) {
-    source = "startup-config";
-    toSet = studyConfig.variation;
-  } else {
-    source = "weightedVariation";
-    // this is the standard arm choosing method
-    const clientId = await studyUtils.getTelemetryId();
-    const hashFraction = await sample.hashFraction(studyConfig.studyName + clientId, 12);
-    toSet = sample.chooseWeighted(studyConfig.weightedVariations, hashFraction);
-  }
-  log.debug(`variation: ${toSet} source:${source}`);
-  return toSet;
-}
-
 function getHomepage(){
   // get the homepage of the user
-  var homepage = Services.prefs.getComplexValue(PREF_HOMEPAGE, Ci.nsIPrefLocalizedString).data;
+  var homepage;
+  try {
+    homepage = Services.prefs.getComplexValue(PREF_HOMEPAGE, Ci.nsIPrefLocalizedString).data;
+  } catch (e) {
+    console.error("Error obtaining the homepage: ", e);
+    return null;
+  }
   // transform the homepage into a nsIURI. Neccesary to get the base domain
-  var homepageURI = Services.netUtils.newURI(homepage);
+  var homepageURI;
+  // TODO: FIX PROBLEM WITH MALFORMED URIS (google.es/)
```

Comment:
This can be handled by the [nsIURIFixup](http://searchfox.org/mozilla-central/rev/e8c36327cd8c9432c69e5e1383156a74330f11f2/docshell/base/nsIURIFixup.idl#121). Something like this should work (please change it to catch errors where needed):

```
let uriFixup = Cc["@mozilla.org/docshell/urifixup;1"].getService(Ci.nsIURIFixup);
Services.eTLD.getBaseDomain(uriFixup .createFixupURI('google.es', Ci.nsIURIFixup.FIXUP_FLAG_NONE));
```

We might just always use the Fixup URI thing instead of calling `newURI` to make things simpler.